src/user.service.ts

```diff
@@ -14,76 +14,80 @@ import { Shop } from './shop.service';
 
 export class UserAddress {
 
-    constructor() {
-    }
-    name: string;
-    note: string;
-    floor: string;
-    streetAdress: string;
-    region: string;
-    postalCode: string;
-    primary: boolean;
-    geo: {
-      lat: number;
-      lng: number;
-    }
+  constructor() {
+  }
```

Comment:
* ici aussi on doit avoir un Object.assign(this,defaultUser,json||{});
* par exemple `geo={}, email={}, addresses=[], roles=[], shops=[], reminder={}, ...` enfin tout pour éviter les nested `?` dans le template

Reply:
haaa okay je savais pas, c'est pour ça que je mettais de ? partout, merci !

Reply:
:+1: